dialects/hir/src/transforms/spill/tests.rs

```diff
@@ -0,0 +1,407 @@
+                AddressSpace::Element,
+            ))))],
+            [AbiParam::new(Type::U32)],
+        ),
+    )?;
+    module.borrow_mut().symbol_manager_mut().insert_new(func, ProgramPoint::Invalid);
+    let callee_sig = Signature::new(
+        [
+            AbiParam::new(Type::Ptr(Arc::new(PointerType::new_with_address_space(
+                Type::U128,
+                AddressSpace::Element,
+            )))),
+            AbiParam::new(Type::U128),
+            AbiParam::new(Type::U128),
+            AbiParam::new(Type::U128),
+            AbiParam::new(Type::U64),
+        ],
+        [AbiParam::new(Type::U32)],
+    );
+    let callee =
+        ob.create_function(Ident::with_empty_span("example".into()), callee_sig.clone())?;
+    module
+        .borrow_mut()
+        .symbol_manager_mut()
+        .insert_new(callee, ProgramPoint::Invalid);
+
+    {
+        let mut b = FunctionBuilder::new(func, &mut ob);
+        let entry = b.current_block();
+        let v0 = entry.borrow().arguments()[0] as ValueRef;
+        let v1 = b.ptrtoint(v0, Type::U32, span)?;
+        let k32 = b.u32(32, span);
+        let v2 = b.add_unchecked(v1, k32, span)?;
+        let v3 = b.inttoptr(
+            v2,
+            Type::Ptr(Arc::new(PointerType::new_with_address_space(
+                Type::U128,
+                AddressSpace::Element,
+            ))),
+            span,
+        )?;
+        let v4 = b.load(v3, span)?;
+        let k64 = b.u32(64, span);
+        let v5 = b.add_unchecked(v1, k64, span)?;
+        let v6 = b.inttoptr(
+            v5,
+            Type::Ptr(Arc::new(PointerType::new_with_address_space(
+                Type::U128,
+                AddressSpace::Element,
+            ))),
+            span,
+        )?;
+        let v7 = b.load(v6, span)?;
+        let v8 = b.u64(1, span);
+        let _ret_from_call = b.exec(callee, callee_sig.clone(), [v6, v4, v7, v7, v8], span)?;
+        let k72 = b.u32(72, span);
+        let v9 = b.add_unchecked(v1, k72, span)?;
+        b.store(v3, v9, span)?;
+        let v10 = b.inttoptr(
+            v9,
+            Type::Ptr(Arc::new(PointerType::new_with_address_space(
+                Type::U64,
+                AddressSpace::Element,
+            ))),
+            span,
+        )?;
+        let _v11 = b.load(v10, span)?;
+        b.ret(Some(v2), span)?;
+    }
+
+    // We expect spills and reloads to be required around the call and later uses:
+    // - At the `exec` call, operand stack pressure forces spilling of two live values.
+    //   The analysis selects the farthest next-use candidates, which here correspond to
+    //   the results used later.
+    // - Before the `store` and `ret` we must reload them back.
+    // After running TransformSpills, those spill/reload pseudos will be materialized as:
+    //   store_local <spilled>
+    //   load_local  : <type> #[local = lvN]
+    let before = func.as_operation_ref().borrow().to_string();
+    expect![[r#"
```

Comment:
We probably should use `expect_file`, because of the issue that arises when multiple `expect` assertions need to be updated concurrently via `UPDATE_EXPECT=1` - it very easily corrupts the file if lines are added/removed as a result of the update, but it's racy, so not something that always happens.

Reply:
Thanks! Done in https://github.com/0xMiden/compiler/pull/636/commits/3464ae706e57e95bbb6b29c056db6bf56be1b3c1